packages/dara-core/dara/core/interactivity/any_variable.py

```diff
@@ -190,35 +190,41 @@ def is_valid(x):
                 raw_results[channel] = e
 
         async with anyio.create_task_group() as tg:
-            for chan in session_channels.values():
-                tg.start_soon(retrieve_value, chan)
+            for channels in session_channels.values():
+                for chan in channels:
+                    tg.start_soon(retrieve_value, chan)
```

Comment:
If we have the channel selected, can we narrow this down to only send requests to that particular channel? And same in the loop processing the results below. We're ignoring other results anyway

We'll save a few requests and then we don't need the extra block below since we already return one value if only one result is present